### DISCUSSION TITLE
Ignore during the build process

### DISCUSSION THREAD

**Discussion by nexxai** [NativePHP/laravel#164] [Ideas]
So I'm working on my build and I've noticed there's a few things that should be added to the build process for cleanliness and security sake, but I don't know enough TS/JS to contribute actual code for them.  I'm hoping someone with a little more knowledge than I can take this ball and run with it.

1. In https://github.com/NativePHP/laravel/commit/15814d8f6347be45ec6cfbdda118959d612bdc93 Marcel added the ability to ignore certain entries from the .env file.  It would be good to have a community curated blacklist of certain common keys that would automatically be excluded to avoid the unintentional disclosure of secure material, with a possible `--i_know_what_i_am_doing` flag to ignore those checks.  (As this is all done in PHP, if the community agrees this is a good idea, I  don't mind being the steward of said list.)
2. Similar to point no. 1, I think a similar feature should be added to allow the ignoring of certain files and even entire folders from the build.  For example: I do not want my local `storage/logs` folder included in my app's build.  My app also uses Ryan Chandler's Orbit package for model storage, so I would like to be able to exclude `/content` from my build as well.
3. My app was hanging with a `500 Server Error` when running as a packaged app but not when running locally.  After some digging into the `~/Library/Application Support/[MY APP]/storage/logs` folder, I discovered it was because I hadn't run `npm run build` prior to the packing of my app.  Having that automatically run that command (or at least prompt for the option) would make my life easier.

I am aware that most of the above issues would be solved by deploying from a fresh CI runner, but I would suspect some smaller devs will be packaging from their local rather than CI (at least in the beginning) and having the above quality of life improvements would be helpful and make the community a little more secure.

**Comment by simonhamp** [comment]
@nexxai I think these are all great ideas. I'm not sure where we are with them now since August, but feel free to dive in if you have time, I'd be happy to work with you to figure some of these out and test
